### PR TITLE
Fix custom taste delete

### PR DESCRIPTION
**Pull request by aLLeXUs**

### CODE REVIEW

File: app/Actions/UserTaste/DeleteTasteAction.php [hunk 1]
```diff
@@ -24,13 +24,12 @@ public function execute(DeleteTasteRequest $deleteTasteRequest)
...
         }
-        if ($taste->user_id !== Auth::id()) {
-            throw new TasteNotFoundException;
-        }
         $this->userRepository->deleteTaste(
             Auth::user(),
             $taste
         );
-        $this->tasteRepository->deleteById($deleteTasteRequest->getTasteId());
+        if ($taste->user_id === Auth::id() && $taste->users()->count() === 0) {
```
Comment (nemoipaha) [inline app/Actions/UserTaste/DeleteTasteAction.php]: $taste->users()->count() === 0 - repository method `tasteRepository->getUsersCount(int $tasteId) || hasUsers($tasteId)`

File: app/Repositories/User/Criterias/GetTastesByUserWithCriteria.php [hunk 2]
```diff
@@ -16,6 +16,12 @@ public function __construct(int $userId)
 
     public function apply($model, RepositoryInterface $repository)
     {
-        return $model->where('user_id', $this->userId)->orWhere('is_default', true);
+        $userId = $this->userId;
```
Comment (nemoipaha) [inline app/Repositories/User/Criterias/GetTastesByUserWithCriteria.php]: remove, $this->user_id will be visible inisde callback

File: app/Repositories/User/Criterias/GetTastesByUserWithCriteria.php [hunk 3]
```diff
@@ -16,6 +16,12 @@ public function __construct(int $userId)
...
+        return $model->whereHas(
+            'users',
+            function ($q) use ($userId) {
```
Comment (nemoipaha) [inline app/Repositories/User/Criterias/GetTastesByUserWithCriteria.php]: types

File: tests/Feature/Api/UserTaste/TasteTest.php [hunk 4]
```diff
@@ -33,7 +33,9 @@ public function test_get_tastes()
     {
         factory(Taste::class, 3)->create([
             'user_id' => $this->user->id
```
Comment (nemoipaha) [inline tests/Feature/Api/UserTaste/TasteTest.php]: 'user_id' => $this->user->id - can be removed?

**Review by nemoipaha — APPROVED** [review]

File: app/Repositories/User/Criterias/GetTastesByUserWithCriteria.php (hunk 2 above)
Comment (aLLeXUs) [reply app/Repositories/User/Criterias/GetTastesByUserWithCriteria.php]: $this->user_id unreachable statement error

File: tests/Feature/Api/UserTaste/TasteTest.php (hunk 4 above)
Comment (aLLeXUs) [reply tests/Feature/Api/UserTaste/TasteTest.php]: we need to assign custom taste to current user, so i think no

**Pull request merged**